src/mem/pagemap.h

```diff
@@ -191,7 +191,7 @@ namespace snmalloc
           return std::pair(nullptr, 0);
 
         shift -= BITS_PER_INDEX_LEVEL;
-        ix = (addr >> shift) & ENTRIES_MASK;
+        ix = (static_cast<size_t>(addr) >> shift) & ENTRIES_MASK;
```

Comment:
I would rather we avoid `static_cast<size_t>` throughout the code and define a function that does this conversion.

We currently have `address_t`, which is actually a pointer and not an address.  We should probably rename that to `pointer_t` or similar and have a separate `address_t`, rather than assuming `size_t` is always the correct thing (`size_t` only has to be big enough for the maximum size of a single object, it would be valid to have a C implementation with 64-bit pointers but a 32-bit `size_t`).

Reply:
On CHERI we have a `vaddr_t` that's guaranteed to be big enough.  I guess we can have the AAL/APL define what `address_t` means?  Is there anywhere we don't want the existing `address_t` to be `uintptr_t`?